File: capstone project/FRONTEND/src/app/services/booking-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/internal/Observable';
import { IBooking } from '../ibooking';
import { IPayment } from '../ipayment';
import { Con1 } from '../services/util/util';

@Injectable({
  providedIn: 'root'
})
export class BookingService {

private baseUrl = `${Con1.apiUrl}/bookings`;

  constructor(private http: HttpClient) {}

  createBooking(booking: IBooking): Observable<IBooking> {
  return this.http.post<IBooking>(`${this.baseUrl}`, booking);
}

  pay(payment: IPayment): Observable<IPayment> {
    return this.http.post<IPayment>(`${this.baseUrl}/pay`, payment);
  }

  downloadTicket(pnr: string): Observable<Blob> {
  return this.http.get(`${this.baseUrl}/${pnr}/pdf`, { responseType: 'blob' });
}
  getBooking(bookingId: number | string): Observable<IBooking> {
  return this.http.get<IBooking>(`${this.baseUrl}/${bookingId}`);
}
}

File: capstone project/Backend/bookingms/src/main/java/com/example/bookingms/service/BookingService.java
package com.example.bookingms.service;
import java.util.Optional;

import com.example.bookingms.dto.Payment;
import com.example.bookingms.model.Booking;


public interface BookingService {
	Booking createBooking(Booking booking);
    Booking getBookingByPnr(String pnr);
    void pay(Payment payment);
    Booking getBookingById(int id);
}